### PR TITLE
Improvements to the SAPM S3 publisher

### PR DESCRIPTION
**Pull request by jmelis**
This PR brings improves several things in the SAPM S3 exporter:

- `app_name` is provided as part of the node. This is necessary because integrations using the publisher data will filter by `app_name`
- Information about whether or not a job is a test job is codified via `publish_job_logs`.
- If target is not defined, it defaults to `None` instead of default. This is more in line with other integrations we have.
- And more importantly, now all nodes that publish will be in the output, which is a changed compared with the current state, where only those that publish AND are being subscribed to by other nodes do appear.

### CODE REVIEW

**Comment by jmelis** [comment]
@fishi0x01 let me know if you prefer the PR to be split into separate ones

**Comment by jmelis** [comment]
To further understand why these changes are needed, it may be interesting to look at https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/102423 (which depends on this MR)

File: reconcile/saas_auto_promotions_manager/utils/saas_files_inventory.py [hunk 1]
```diff
@@ -34,6 +34,11 @@ def __init__(self, saas_files: Iterable[SaasFile]):
         self._assemble_publishers()
         self._remove_unsupported()
 
+
+    @property
+    def publishers_with_subscribers(self) -> list[Publisher]:
```
Comment (jmelis) [inline reconcile/saas_auto_promotions_manager/utils/saas_files_inventory.py:39]: @fishi0x01 note that I added this to keep tests adequate.

This changes things then, and I'm not sure I understand the consequences. Before this PR `publishers` report nodes that have subscribers. After this PR is `publishers` will report all nodes that publish, and you need to specifically access `publishers_with_subscribers` if you want to get to the previous behaviour.

Alternatively we could reverse it, i.e. keep `publishers` as it was, add `publishers_without_subscribers`, and leverage that one in the S3 exporter.

What do you prefer?

**Review by fishi0x01 — APPROVED** [review]
LGTM

**Pull request merged**